daphne/src/roles.rs

```diff
@@ -582,14 +580,33 @@ pub trait DapHelper<S>: DapAggregator<S> {
                     ));
                 }
 
+                // Remove reports that are rejected early.
+                let early_rejects = early_rejects_future.await?;
                 let agg_resp = match transition {
-                    DapHelperTransition::Continue(state, agg_resp) => {
+                    DapHelperTransition::Continue(mut state, mut agg_resp) => {
+                        let mut i = 0;
+                        while i < state.seq.len() {
+                            if let Some(failure) = early_rejects.get(&agg_resp.transitions[i].nonce)
+                            {
+                                // Mark reports that were rejected early as TransitionVar::Failed.
+                                agg_resp.transitions[i].var = TransitionVar::Failed(*failure);
+
+                                // Remove VdafState of reports that were rejected early.
+                                if state.seq[i].1 == agg_resp.transitions[i].nonce {
```

Comment:
If the nonces *don't* match here, then this is a pretty serious bug. (These nonce sequences are supposed to be the same, as enforced by tests in `vdaf/mod_test.rs`.) If this condition is false, then we should abort with an internal error.

Reply:
Sounds good, I've added an else statement that responds with and error if the nonces don't match.